src/state.rs

```diff
@@ -123,12 +246,22 @@ pub struct UiState {
   pub queries: Vec<(Query, Option<usize>)>,
   /// The currently selected `Query`.
   pub selected: Option<usize>,
+  /// The configured colors
+  pub colors: Option<Colors>,
 }
 
 impl UiState {
   /// Persist the state into a file.
   pub fn save(&self) -> Result<()> {
-    save_state(&self.path, self.to_serde())
+    let ui_state = load_state::<SerUiState>(self.path.as_ref()).unwrap_or_default();
+
+    let state_to_save = UiState {
+      path: PathBuf::from(self.path.as_path()),
+      queries: self.queries.clone(),
+      selected: self.selected,
+      colors: Some(ui_state.colors),
+    };
+    save_state(&self.path, state_to_save.to_serde())
```

Comment:
Mind adding a comment explaining why we need this hocus-pocus? :)

Hm, wouldn't it be better to just overwrite `self.colors` instead of cloning so much state? Or do you see a problem with that?

Reply:
I tried to just overwrite `self.colors`, but that's an immutable reference (`&self`), so the compiler wouldn't let me. I can add a comment describing what's going on here.

Or do you have an idea how we can overwrite `self.colors` here?